copylot/hardware/cameras/abstract_camera.py

```diff
@@ -1,5 +1,113 @@
-import abc
+from abc import ABCMeta, abstractmethod
 
 
-class AbstractCamera(metaclass=abc.ABCMeta):
-    pass
+class AbstractCamera(metaclass=ABCMeta):
+
+    @abstractmethod
+    def snap(self):
+        """Method to capture a single image from the camera."""
+        raise NotImplementedError()
+
+    @abstractmethod
+    def list_available_cameras(self):
+        """List all cameras that driver discovers."""
+        raise NotImplementedError()
+
+    @property
+    @abstractmethod
+    def exposure(self):
+        raise NotImplementedError()
+
+    @exposure.setter
+    @abstractmethod
+    def exposure(self, value):
+        raise NotImplementedError()
+
+    @property
+    @abstractmethod
+    def exposure_limits(self):
+        """
+        Valid minimum and maximum exposure values.
+
+        Returns
+        -------
+        Tuple
+            (min_valid_exposure, max_valid_exposure)
+        """
+        raise NotImplementedError()
+
+    @property
+    @abstractmethod
+    def bitdepth(self):
+        raise NotImplementedError()
+
+    @bitdepth.setter
+    @abstractmethod
+    def bitdepth(self, value):
+        raise NotImplementedError()
+
+    @property
+    @abstractmethod
+    def image_size(self):
+        raise NotImplementedError()
+
+    @image_size.setter
+    @abstractmethod
+    def image_size(self, value):
+        raise NotImplementedError()
```

Comment:
should it be clarified here that `value` must be a tuple?

Reply:
hard to say at  this point, I rather wait and see APIs from more camera vendors on this then  decide on what type to enforce in abstract camera class

Reply:
I think here we should pass a tuple and the function can handle those as needed depending on vendors. Most likely we would need 2dimensions anyways.

Reply:
okay then I will go with tuple for now